src/Common/src/Interop/FreeBSD/Interop.libutil.cs

```diff
@@ -0,0 +1,429 @@
+            long    ki_flag;                    /* P_* flags */
+            long    ki_kiflag;                  /* KI_* flags (below) */
+            int     ki_traceflag;               /* Kernel trace points */
+            byte    ki_stat;                    /* S* process status */
+            public byte ki_nice;                /* Process "nice" value */
+            byte    ki_lock;                    /* Process lock (prevent swap) count */
+            byte    ki_rqindex;                 /* Run queue index */
+            byte  ki_oncpu_old;                 /* Which cpu we are on (legacy) */
+            byte  ki_lastcpu_old;               /* Last cpu we were on (legacy) */
+            public fixed byte    ki_tdname[TDNAMLEN+1]; /* thread name */
+            fixed byte    ki_wmesg[WMESGLEN+1];         /* wchan message */
+            fixed byte    ki_login[LOGNAMELEN+1];       /* setlogin name */
+            fixed byte    ki_lockname[LOCKNAMELEN+1];   /* lock name */
+            public fixed byte    ki_comm[COMMLEN+1];    /* command name */
+            fixed byte    ki_emul[KI_EMULNAMELEN+1];    /* emulation name */
+            fixed byte    ki_loginclass[LOGINCLASSLEN+1]; /* login class */
+            fixed byte    ki_sparestrings[50]; /* spare string space */
+            fixed int ki_spareints[KI_NSPARE_INT];    /* spare room for growth */
+            int ki_oncpu;       /* Which cpu we are on */
+            int ki_lastcpu;     /* Last cpu we were on */
+            int ki_tracer;      /* Pid of tracing process */
+            int ki_flag2;       /* P2_* flags */
+            int ki_fibnum;      /* Default FIB number */
+            uint   ki_cr_flags;        /* Credential flags */
+            int ki_jid;         /* Process jail ID */
+            public int ki_numthreads;      /* XXXKSE number of threads in total */
+            public int ki_tid;         /* XXXKSE thread id */
+            fixed byte ki_pri[4];    /* process priority */
+            public rusage ki_rusage;   /* process rusage statistics */
+            /* XXX - most fields in ki_rusage_ch are not (yet) filled in */
+            rusage ki_rusage_ch;    /* rusage of children processes */
+            void *ki_pcb;        /* kernel virtual addr of pcb */
+            void    *ki_kstack;     /* kernel virtual addr of stack */
+            void    *ki_udata;      /* User convenience pointer */
+            public void *ki_tdaddr;  /* address of thread */
+
+            fixed long ki_spareptrs[KI_NSPARE_PTR];   /* spare room for growth */
+            fixed long    ki_sparelongs[KI_NSPARE_LONG];  /* spare room for growth */
+            long    ki_sflag;       /* PS_* flags */
+            public long    ki_tdflags;     /* XXXKSE kthread flag */
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct sysctl_name
+        {
+            public fixed int name[4];
+        }
+
+        [DllImport("libc", SetLastError = true)]
```

Comment:
As noted above, in our production code we've avoided P/Invoking directly into libc.  We also define the libraries we P/Invoke into in the Interop.Libraries.cs files and use the consts here rather than the library names directly.